File: Python/recursion/coin_change.py
def get_count(coins, amt):
    def _get_count(coins, n, amt):
        if amt == 0:
            return 1
        if n == 0:
            return 0
        count = _get_count(coins, n - 1, amt)
        if coins[n - 1] <= amt:
            count += _get_count(coins, n, amt - coins[n - 1])
        return count

    return _get_count(coins, len(coins), amt)


if __name__ == "__main__":
    print(get_count([2, 3, 5, 6], 10))

File: C++/Recursion/CoinChange.cpp
#include <bits/stdc++.h>
using namespace std;

int getCount(int coins[], int n, int amount)
{
  if (amount == 0)
    return 1;
  if (n == 0)
    return 0;
  int count = getCount(coins, n - 1, amount);
  if (coins[n - 1] <= amount)
    count += getCount(coins, n, amount - coins[n - 1]);
  return count;
}

// int main()
// {
//   int coins[] = {2, 3, 5, 6},
//       n = sizeof(coins) / sizeof(coins[0]);
//   cout << getCount(coins, n, 10) << endl;
// }
